src/webui/api/authcontroller.cpp

```diff
@@ -45,11 +45,15 @@ AuthController::AuthController(ISessionManager *sessionManager, IApplication *ap
 
 void AuthController::setUsername(const QString &username)
 {
+    if (!m_username.isEmpty() && (m_username != username))
```

Comment:
Well, I've reviewed it in more detail and I'm convinced that you chose the wrong place to apply it. I believe it should be placed directly in WebApplication:

https://github.com/qbittorrent/qBittorrent/blob/0ea35c54a347bbfddf97f0b61982416f27ee52b4/src/webui/webapplication.cpp#L302-L305
https://github.com/qbittorrent/qBittorrent/blob/0ea35c54a347bbfddf97f0b61982416f27ee52b4/src/webui/webapplication.cpp#L307-L310